### PR TITLE
enh(api): implements infrastructure for find open api use case

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

this PR intends to implements infrastructure for find open api use case

**Fixes** # MON-12484

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: src/Core/Application/Security/ProviderConfiguration/OpenId/UseCase/FindOpenIdConfiguration/FindOpenIdConfiguration.php [hunk 1]
```diff
@@ -63,6 +63,24 @@ public function __invoke(FindOpenIdConfigurationPresenterInterface $presenter):
...
+        $findOpenIdConfigurationResponse = new FindOpenIdConfigurationResponse();
+        $findOpenIdConfigurationResponse->isActive = $configuration->isActive();
+        $findOpenIdConfigurationResponse->isForced = $configuration->isForced();
+        $findOpenIdConfigurationResponse->trustedClientAddresses = $configuration->getTrustedClientAddresses();
+        $findOpenIdConfigurationResponse->blacklistClientAddresses = $configuration->getBlacklistClientAddresses();
+        $findOpenIdConfigurationResponse->baseUrl = $configuration->getBaseUrl();
+        $findOpenIdConfigurationResponse->authorizationEndpoint = $configuration->getAuthorizationEndpoint();
+        $findOpenIdConfigurationResponse->tokenEndpoint = $configuration->getTokenEndpoint();
+        $findOpenIdConfigurationResponse->introspectionTokenEndpoint = $configuration->getIntrospectionTokenEndpoint();
+        $findOpenIdConfigurationResponse->userInformationsEndpoint = $configuration->getUserInformationsEndpoint();
```
Comment (kduret) [inline src/Core/Application/Security/ProviderConfiguration/OpenId/UseCase/FindOpenIdConfiguration/FindOpenIdConfiguration.php:75]:
```suggestion
        $findOpenIdConfigurationResponse->userInformationEndpoint = $configuration->getUserInformationEndpoint();
```

File: src/Core/Infrastructure/Security/ProviderConfiguration/OpenId/Api/FindOpenIdConfiguration/FindOpenIdConfigurationPresenter.php [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        $presenterResponse = [
+            'is_active' => $response->isActive,
+            'is_forced' => $response->isForced,
+            'trusted_client_addresses' => $response->trustedClientAddresses,
+            'blacklist_client_addresses' => $response->blacklistClientAddresses,
+            'base_url' => $response->baseUrl,
+            'authorization_endpoint' => $response->authorizationEndpoint,
+            'token_endpoint' => $response->tokenEndpoint,
+            'introspection_token_endpoint' => $response->introspectionTokenEndpoint,
+            'userinfo_endpoint' => $response->userInformationsEndpoint,
```
Comment (kduret) [inline src/Core/Infrastructure/Security/ProviderConfiguration/OpenId/Api/FindOpenIdConfiguration/FindOpenIdConfigurationPresenter.php:49]:
```suggestion
            'userinfo_endpoint' => $response->userInformationEndpoint,
```

File: src/Core/Infrastructure/Security/ProviderConfiguration/OpenId/Api/UpdateOpenIdConfiguration/UpdateOpenIdConfigurationController.php [hunk 3]
```diff
@@ -72,8 +72,7 @@ private function createUpdateOpenIdConfigurationRequest(Request $request): Updat
         $updateOpenIdConfigurationRequest->userInformationsEndpoint = $requestData['userinfo_endpoint'];
```
Comment (kduret) [inline src/Core/Infrastructure/Security/ProviderConfiguration/OpenId/Api/UpdateOpenIdConfiguration/UpdateOpenIdConfigurationController.php:72]:
```suggestion
        $updateOpenIdConfigurationRequest->userInformationEndpoint = $requestData['userinfo_endpoint'];
```

File: src/Core/Infrastructure/Security/ProviderConfiguration/OpenId/Repository/DbWriteOpenIdConfigurationRepository.php [hunk 4]
```diff
@@ -81,7 +81,7 @@ private function buildCustomConfigurationFromOpenIdConfiguration(OpenIdConfigura
             'token_endpoint' => $configuration->getTokenEndpoint(),
             'introspection_token_endpoint' => $configuration->getIntrospectionTokenEndpoint(),
             'userinfo_endpoint' => $configuration->getUserInformationsEndpoint(),
```
Comment (kduret) [inline src/Core/Infrastructure/Security/ProviderConfiguration/OpenId/Repository/DbWriteOpenIdConfigurationRepository.php:83]:
```suggestion
            'userinfo_endpoint' => $configuration->getUserInformationEndpoint(),
```

File: tests/php/Core/Application/Security/ProviderConfiguration/OpenId/UseCase/FindOpenIdConfiguration/FindOpenIdConfigurationTest.php [hunk 5]
```diff
@@ -87,7 +87,22 @@ public function testFindConfiguration(): void
...
+        $this->assertTrue($presenter->response->isActive);
+        $this->assertTrue($presenter->response->isForced);
+        $this->assertFalse($presenter->response->verifyPeer);
+        $this->assertEquals([], $presenter->response->trustedClientAddresses);
+        $this->assertEquals([], $presenter->response->blacklistClientAddresses);
+        $this->assertEquals('http://127.0.0.1/auth/openid-connect', $presenter->response->baseUrl);
+        $this->assertEquals('/authorization', $presenter->response->authorizationEndpoint);
+        $this->assertEquals('/token', $presenter->response->tokenEndpoint);
+        $this->assertEquals('/introspect', $presenter->response->introspectionTokenEndpoint);
+        $this->assertEquals('/userinfo', $presenter->response->userInformationsEndpoint);
```
Comment (kduret) [inline tests/php/Core/Application/Security/ProviderConfiguration/OpenId/UseCase/FindOpenIdConfiguration/FindOpenIdConfigurationTest.php:99]:
```suggestion
        $this->assertEquals('/userinfo', $presenter->response->userInformationEndpoint);
```

File: tests/php/Core/Infrastructure/Security/ProviderConfiguration/OpenId/Api/FindOpenIdConfiguration/FindOpenIdConfigurationControllerTest.php [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+
+
+declare(strict_types=1);
```
Comment (kduret) [inline tests/php/Core/Infrastructure/Security/ProviderConfiguration/OpenId/Api/FindOpenIdConfiguration/FindOpenIdConfigurationControllerTest.php:26]:
```suggestion
```

File: src/Core/Infrastructure/Security/ProviderConfiguration/OpenId/Repository/DbReadOpenIdConfigurationRepository.php [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+use Centreon\Infrastructure\DatabaseConnection;
+use Centreon\Infrastructure\Repository\AbstractRepositoryDRB;
+use Core\Application\Security\ProviderConfiguration\OpenId\Repository\ReadOpenIdConfigurationRepositoryInterface
+    as ReadRepositoryInterface;
+use Core\Domain\Security\ProviderConfiguration\OpenId\Model\OpenIdConfiguration;
+
+class DbReadOpenIdConfigurationRepository extends AbstractRepositoryDRB implements ReadRepositoryInterface
+{
+    public function __construct(DatabaseConnection $db)
```
Comment (kduret) [inline src/Core/Infrastructure/Security/ProviderConfiguration/OpenId/Repository/DbReadOpenIdConfigurationRepository.php:34]: missing phpdoc

**Review by kduret — APPROVED** [review]

**Review by a-launois — APPROVED** [review]

**Pull request merged**